v03_pipeline/lib/reference_data/compare_globals.py

```diff
@@ -15,131 +19,125 @@
-        ),
-    }
-
-    results = []
-    for check, result in checks.items():
-        if result is False:
-            logger.info(f'{check} mismatch for {dataset}')
-        results.append(result)
-    return all(results)
-
-
-def ht_version_matches_config(
-    joined_ht_globals: ReferenceDataGlobals,
-    dataset: str,
-    dataset_config: dict,
-    dataset_ht: hl.Table,
-) -> bool:
-    joined_ht_version = joined_ht_globals.versions.get(dataset)
-    if joined_ht_version is None:
-        return False
-
-    config_or_dataset_version = hl.eval(
-        parse_dataset_version(
-            dataset_ht,
-            dataset,
-            dataset_config,
-        ),
-    )
-    return joined_ht_version == config_or_dataset_version
-
-
-def ht_path_matches_config(
-    joined_ht_globals: ReferenceDataGlobals,
-    dataset: str,
-    dataset_config: dict,
-) -> bool:
-    joined_ht_path = joined_ht_globals.paths.get(dataset)
-    if joined_ht_path is None:
-        return False
-
-    config_path = get_ht_path(dataset_config)
-    return joined_ht_path == config_path
-
-
-def ht_enums_match_config(
-    joined_ht_globals: ReferenceDataGlobals,
-    dataset: str,
-    dataset_config: dict,
-) -> bool:
-    joined_ht_enums = joined_ht_globals.enums.get(dataset, {})
-    config_enums = dataset_config.get('enum_select', {})
-    return joined_ht_enums == config_enums
-
-
-def ht_selects_match_config(
-    joined_ht: hl.Table,
-    dataset: str,
-    dataset_config: dict,
-    dataset_ht: hl.Table,
-) -> bool:
-    joined_ht_selects = set(joined_ht[dataset])
-    config_selects = set(get_all_select_fields(dataset_ht, dataset_config).keys())
-    return len(config_selects.symmetric_difference(joined_ht_selects)) == 0
+    selects: dict[str, set[str]]
+
+    @classmethod
+    def from_dataset_configs(
+        cls,
+        rdc: ReferenceDatasetCollection,
+        dataset_type: DatasetType,
+        reference_genome: ReferenceGenome,
+    ):
+        paths, versions, enums, selects = {}, {}, {}, {}
+        for dataset in rdc.datasets(dataset_type):
+            dataset_config = CONFIG[dataset][reference_genome.v02_value]
+            dataset_ht = import_ht_from_config_path(dataset_config, reference_genome)
+
+            paths[dataset] = get_ht_path(dataset_config)
+            versions[dataset] = hl.eval(
+                parse_dataset_version(
+                    dataset_ht,
+                    dataset,
+                    dataset_config,
+                ),
+            )
+            enums[dataset] = dataset_config.get('enum_select', {})
+            selects[dataset] = set(
+                get_all_select_fields(dataset_ht, dataset_config).keys(),
+            )
+        return cls(paths, versions, enums, selects)
+
+    @classmethod
+    def from_ht(
+        cls,
+        ht: hl.Table,
+        rdc: ReferenceDatasetCollection,
+        dataset_type: DatasetType,
+    ):
+        rdc_globals_struct = hl.eval(ht.index_globals())
```

Comment:
I think you maybe only need `hl.eval(ht.globals)` here, or `ht.globals.collect()[0]`.